### DISCUSSION TITLE
Using Firebase Storage

### DISCUSSION THREAD

**Discussion by MattHeslington** [gladly-team/next-firebase-auth#266] [Q&A] [answered]
Hello Gladly,

Is there any known issue regarding using Firebase Storage? I cannot get my app to recognise the storageBucket and keep getting 

> "FirebaseError: Firebase Storage: No default bucket found. Did you set the 'storageBucket' property when initializing the app?"

I've tried copying in the Firebase config directly to my Firebase init file because I couldn't understand why it won't recognise the storageBucket (i.e. not using process.env...) but it still wont recognise it. I've built lots of projects with Firebase and have never experienced this before. I have followed the instructions from [https://github.com/gladly-team/next-firebase-auth#using-the-firebase-apps](https://github.com/gladly-team/next-firebase-auth#using-the-firebase-apps):

`import firebase from 'firebase/app'`
`import 'firebase/storage'`

`const account = () => {`
    `const storageRef = firebase.storage().ref("avatars")`
    `const fileRef = storageRef.child(`${nanoid()}`)`

I've even tried moving over to Supabase storage, but the app wont recognise the supabase instance either, but maybe that could be a different issue.

It's a private account page, so I'm closing out the page with:

`export const getServerSideProps = withAuthUserTokenSSR({ whenUnauthed: AuthAction.REDIRECT_TO_LOGIN })()`

`const AccountWithAuth = withAuthUser({`
    `whenUnauthedAfterInit: AuthAction.REDIRECT_TO_LOGIN,`
`})(account)`

`AccountWithAuth.layout = Layout`

`export default AccountWithAuth`

Many thanks for any guidance.
Kind regards, Matt

**Comment by shreyas-jadhav** [comment, marked as answer]
I had the same issue for a while. Passing the bucket URL in the `firebase.app().storage('url')` function worked for me 😬

```js
const storage = firebase.app().storage('gs://your-project.appspot.com');
        
const storageRef = await storage.ref();

```

**Reply by MattHeslington** [reply]
This worked perfectly. You're an absolute star, thank you

**Reply by fsolarv22** [reply]
Same problem here. any idea how to solve it with firebase v9 and next-firebase-auth@canary?

**Reply by im-ghost** [reply]
When initializing the app,I already put the necessary configurations and it worked perfectly not until I put the keys as an environmental variable

**Comment by kmjennison** [comment]
No known bugs, but please let me know if you identify any problems. FYI, the `firebaseClientInitConfig` config value is [passed directly to `firebase.initializeApp`](https://github.com/gladly-team/next-firebase-auth/blob/main/src/initFirebaseClientSDK.js#L13).
